### DISCUSSION TITLE
Way to test more than visual testing

### DISCUSSION THREAD

**Discussion by AmitSheer** [networkx/networkx#5577] [Q&A] [answered]
Hello, I'm trying to implement a new layout algorithm. I want to implement test for the algorithm, is there a way to implement tests, that are not dependent on visualization?

**Comment by dschult** [comment, marked as answer]
Layout algorithms should focus on the position dict they create. So, you should test the position values in that dict which result from the algorithm.  That will have nothing to do with actually drawing the network.

You might take a look at the tests in [drawing/tests/test_layout.py](https://github.com/networkx/networkx/blob/main/networkx/drawing/tests/test_layout.py).
But many of them don't actually test the values -- rather they just make sure the function doesn't raise an exception. It would be better to test an example where the resulting positions are known -- even if they are known only from running the code previously. That way we at least know when changes in the code change the positions. :}
